runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,585 @@
+		}
+	}
+
+	containerId, err := c.createScanContainer(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create scan container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	err = c.dockerClient.ContainerStart(ctx, containerId, types.ContainerStartOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c Client) createScanContainer(ctx context.Context, config *provider.ScanJobConfig) (string, error) {
+	// Check if scan container already created
+	scanName, err := getScanName(config)
+	if err != nil {
+		return "", provider.FatalError{Err: err}
+	}
+	containerId, err := c.getContainerIdFromContainerName(ctx, scanName)
+	if containerId != "" {
+		return containerId, nil
+	}
+
+	pl, err := c.dockerClient.ImagePull(ctx, config.ScannerImage, types.ImagePullOptions{})
+	if err != nil {
+		return "", err
+	}
+	_, _ = io.Copy(io.Discard, pl)
+	_ = pl.Close()
+
+	scanConfigFilePath, err := getScanConfigFileName(config)
+	if err != nil {
+		return "", err
+	}
+
+	script := fmt.Sprintf(`
+/app/vmclarity-cli \
+--config /tmp/%s \
+--server %s \
+--asset-scan-id %s
+`, filepath.Base(scanConfigFilePath), config.VMClarityAddress, config.AssetScanID)
+
+	resp, err := c.dockerClient.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image:      config.ScannerImage,
+			Entrypoint: []string{"sh", "-c", script},
+		},
+		&container.HostConfig{
+			Binds: []string{fmt.Sprintf("%s:/tmp", path.Dir(scanConfigFilePath))},
+			Mounts: []mount.Mount{
+				{
+					Type:   mount.TypeVolume,
+					Source: scanName,
+					Target: "/mnt/snapshot",
+				},
+			},
+		},
+		nil,
+		nil,
+		scanName,
+	)
+	if err != nil {
+		return "", err
+	}
+
+	return resp.ID, nil
+}
+
+func (c Client) getContainerIdFromContainerName(ctx context.Context, scanName string) (string, error) {
+	filters := filters.NewArgs()
+	filters.Add("name", scanName)
+	containers, err := c.dockerClient.ContainerList(ctx, types.ContainerListOptions{
+		All:     true,
+		Filters: filters,
+	})
+	if err != nil {
+		return "", provider.FatalError{
+			Err: fmt.Errorf("failed to get containers. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	if len(containers) == 0 {
+		return "", provider.FatalError{
+			Err: fmt.Errorf("scan container not found. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	if len(containers) > 1 {
+		return "", provider.FatalError{
+			Err: fmt.Errorf("found more than one scan container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	return containers[0].ID, nil
+}
+
+func (c Client) createScanConfigFile(config *provider.ScanJobConfig) error {
+	scanConfigFilePath, err := getScanConfigFileName(config)
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(scanConfigFilePath); errors.Is(err, os.ErrNotExist) {
+		err = os.WriteFile(scanConfigFilePath, []byte(config.ScannerCLIConfig), 0644)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Non `os.ErrNotExist` errors are not handled (unsure if it can occur, but either way should be handled too)